tests/potential/test_potential.py

```diff
@@ -308,5 +309,281 @@ def test_fromfile(self):
+
+        #test creation with all params, default values
+        p = test_PairPotential.LinPot(types=('1',), params=('m','rmin','rmax','shift'),
+                                      default={'m':3.5,'rmin':0.0,'rmax':1.0,'shift':True})
+        coeff = potential.CoefficientMatrix(types=('1',), params=('m','rmin','rmax','shift'),
+                                            default={'m':3.5,'rmin':0.0,'rmax':1.0,'shift':True})
+        self.assertCountEqual(p.coeff.types, coeff.types)
+        self.assertCountEqual(p.coeff.params, coeff.params)
+        self.assertDictEqual(p.coeff.default.todict(), coeff.default.todict())
+
+        #test creation with all params, no default values
+        p = test_PairPotential.LinPot(types=('1',), params=('m',))
+        coeff = potential.CoefficientMatrix(types=('1',), params=('m','rmin','rmax','shift'),
+                                            default={'rmin':False,'rmax':False,'shift':False})
+        self.assertCountEqual(p.coeff.types, coeff.types)
+        self.assertCountEqual(p.coeff.params, coeff.params)
+        self.assertDictEqual(p.coeff.default.todict(), coeff.default.todict())
+
+    def test_zeros(self):
+        """Test _zeros method"""
+        p = test_PairPotential.LinPot(types=('1',), params=('m',))
+
+        #test with scalar r
+        r = 0.5
+        r_copy, u, s = p._zeros(r)
+        self.assertEqual(r_copy, r)
+        self.assertEqual(u, 0)
+        self.assertEqual(s, True)
+
+        u_copy = np.zeros(2)
+        #test with list r
+        r = [0.2, 0.3]
+        r_copy, u, s = p._zeros(r)
+        np.testing.assert_allclose(r_copy, r)
+        np.testing.assert_allclose(u, u_copy)
+        self.assertEqual(s, False)
+
+        #test with tuple r
+        r = (0.2, 0.3)
+        r_copy, u, s = p._zeros(r)
+        np.testing.assert_allclose(r_copy, r)
+        np.testing.assert_allclose(u, u_copy)
+        self.assertEqual(s, False)
+
+        #test with numpy array r
+        r = np.array([0.2, 0.3])
+        r_copy, u, s = p._zeros(r)
+        np.testing.assert_allclose(r_copy, r)
+        np.testing.assert_allclose(u, u_copy)
+        self.assertEqual(s, False)
+
+        #test with non 1-d array r
+        r = np.array([[1, 2], [0.2, 0.3]])
+        with self.assertRaises(TypeError):
+            r_copy, u, s = p._zeros(r)
+
+    def test_energy(self):
+        """Test energy method"""
+        #test with no cutoffs
+        p = test_PairPotential.LinPot(types=('1',), params=('m',),
+                                      default={'m':2})
+        u = p.energy(pair=(('1','1')), r=0.5)
+        self.assertEqual(u, 1.0)
+        u = p.energy(pair=(('1','1')), r=[0.25,0.75])
+        np.testing.assert_allclose(u, [0.5,1.5])
+
+        #test with rmin set
+        p = test_PairPotential.LinPot(types=('1',), params=('m','rmin'),
+                                      default={'m':2,'rmin':0.0})
+        u = p.energy(pair=(('1','1')), r=0.0)
```

Comment:
Make `rmin` something bigger than 0.0 like 0.1, it isn't a good test because the potential is trivially `u(0) = 0` even if you don't use `rmin` correctly.